rust/services/workload/src/lib.rs

```diff
@@ -219,41 +235,84 @@ impl WorkloadApi {
     // NB: Automatically published by the nats-db-connector
     pub async fn handle_db_update(&self, msg: Arc<Message>) -> Result<types::ApiResult, anyhow::Error> {
         log::debug!("Incoming message for 'WORKLOAD.update'");
         
         let payload_buf = msg.payload.to_vec();
-        let workload: schemas::Workload = serde_json::from_slice(&payload_buf)?;
-        log::trace!("New workload to assign. Workload={:#?}", workload);
+
+        let mut workload: schemas::Workload = serde_json::from_slice(&payload_buf)?;
+        log::trace!("New workload to assign. Workload={:#?}", workload.clone());
+
+        // 1. fetch existing workload document
+        match self.workload_collection.get_one_from(doc! { "_id":  workload._id.clone() }).await?{
+            Some(workload_doc) => {
+                // 2. remove workload from hosts
+                self.host_collection.update_one_within(
+                    doc!{ "_id": { "$in": workload_doc.assigned_hosts } },
+                    UpdateModifications::Document(doc!{ "$pull": { "assigned_workloads": workload_doc._id.clone() } })
+                ).await?;
+                
+                // 3. remove workload from developers
+                self.developer_collection.update_one_within(
+                    doc!{ "_id": { "$in": workload_doc.assigned_developer } },
+                    UpdateModifications::Document(doc!{ "$pull": { "assigned_workloads": workload_doc._id.clone() } })
+                ).await?;
+            },
+            None => {
+                log::warn!("Workload not found. Unable to remove workload.");
+            }
+        }
+
+        // 4. add workload to hosts
+        self.host_collection.update_one_within(
+            doc!{ "_id": { "$in": workload.clone().assigned_hosts } },
+            UpdateModifications::Document(doc!{ "$push": { "assigned_workloads": workload._id.clone() } })
+        ).await?;
         
-        // TODO: ...handle the use case for the update entry change stream 
+        // 5. add workload to developers
+        self.developer_collection.update_one_within(
+            doc!{ "_id": { "$in": workload.clone().assigned_developer } },
+            UpdateModifications::Document(doc!{ "$push": { "assigned_workloads": workload._id.clone() } })
+        ).await?;
+
+        workload.updated_at = Some(DateTime::from_chrono(chrono::Utc::now()));
+        let updated_workload_doc = to_document(&workload)?;
+        self.workload_collection.update_one_within(
+            doc! { "_id": workload._id.clone() },
+            UpdateModifications::Document(doc!{ "$set": updated_workload_doc }),
+        ).await?;
 
         let success_status = WorkloadStatus {
-            id: workload._id,
+            id: match workload._id.clone() {
+                Some(oid) => Some(oid.to_hex()),
+                None => None
+            },
             desired: WorkloadState::Running,
             actual: WorkloadState::Running,
```

Comment:
```suggestion
            desired: WorkloadState::Updating,
            actual: WorkloadState::Updating,
```
Can we add a state to represent the state wherein a workload will need the host to update a workload?